src/search/landmarks/landmark_count_heuristic.cc

```diff
@@ -263,20 +254,48 @@ bool LandmarkCountHeuristic::generate_helpful_actions(
 }
 
 bool LandmarkCountHeuristic::landmark_is_interesting(
-    const State &state, const LandmarkNodeSet &reached, LandmarkNode &lm_node) const {
-    /* A landmark is interesting if it hasn't been reached before and
-     its parents have all been reached, or if all landmarks have been
-     reached before, the LM is a goal, and it's not true at moment */
-
-    int num_reached = reached.size();
-    if (num_reached != lgraph->get_num_landmarks()) {
-        if (reached.find(&lm_node) != reached.end())
-            return false;
-        else
-            return !check_node_orders_disobeyed(lm_node, reached);
+    const State & /*state*/, LandmarkNode &lm_node) const {
+    /*
+      A landmark is interesting if it hasn't been reached before and
+      its parents have all been reached, or if all landmarks have been
+      reached before, the LM is a goal, and it's not true at the moment.
+    */
```

Comment:
This comment is not true anymore. Either remove it or (better) replace it with something like this:
A landmark is interesting if it must be achieved in the future and all its parents have been reached.
I would then also remove the comment for the `lm_not_reached` case below